x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -47,17 +19,16 @@ thread_local! {
 /// Returns an [`StateError`] if the pointer or length of `args` exceeds
 /// the maximum size of a u32.
 #[allow(clippy::cast_possible_truncation)]
-pub fn to_host_ptr(arg: &[u8]) -> Result<HostPtr, StateError> {
-    let ptr = arg.as_ptr() as usize;
+pub fn to_ffi_ptr(arg: &[u8]) -> Result<CPointer, StateError> {
```

Comment:
Can do this in a follow up since it would be a small change, but shouldn't the `CPointer` be defined in this module? Pointers are related to `memory`, not really a `program`.

Reply:
I updated it in this PR!